mit-bitcoin-expo/mit-bitcoin-expo-2021/defending-against-99-attack.md

```diff
@@ -0,0 +1,76 @@
+
+So we can sort of mix these, the mining proof of work and prevention of the collision attacks. So you can say you're going to put the block hash confirming a transaction into the UTXO tree. So now you have this commitment to the proof of work. It also commits to its position in the tree so that the attacker can't collide with an arbitrary element. So for example, the hashes are now the number two (the value), the block hash that confirms it and the UTXO. So this makes collision attacks much harder, because instead of performing a single hash to try and find the collision, you need to mine a block for every attempt. And you can't reuse this work, everytime you try one it changes the block itself. You have two merkle trees, one that has the block hash and one that has the UTXO set. Now you've changed the amount of work you have to do. It's still a collision attack, it's still 2^128 operations, but the operation has changed from a hash function to mining an entire block, which currently is something like 2^70 hashes. So that is a huge increase. Even though in some sense its the same equation. 
+
+So if it's 32 byte hashes, you'd need to mine 2^128 blocks to collide, or to do a preimgae attack you'de need to do 2^256 hashes. If you reduced your hash to 16 bytes, you'd need to mine 2^64 blocks to collide, or 2^128 hahes for a preimage attack. And 2^64 blocks is a lot of blocks, 18 quintillion blocks. So it does seem like yeah, it's a reduction in security, but really are you meaningfully defending against an attacker that can mine 18 quintillion blocks? 
+
+So that's the general idea here. Do we need to reduce hash sizes for these things? No. Is too much security a waste? It can be. It can be very efficient to reduce the size of these things by half. To reduce downloads, and reduce disk storgage by half, and speed things up. That's a meaningful gain. Maybe addresses are a little longer than they need to be, but they're not twice as long as they need to be. This is an interesting idea, so it's something I want to put out there. It's something we should thinkg about. Maybe there are other places that people are working on where there's proof of work happening. The Bitcoin blockchain keeps going on, doing enormous amounts of work and maybe we're wasting some of that work. Maybe we can use it to increase efficiency. If you're been looking at the news in the last 6 months since Bitcoin's been going up, a lot of people think Bitcoin is really wasteful and it's pointless. I don't think they're right about it. But are we using this proof of work? It's happening. Are we taking full advantage of this? Can we leverage it for more efficiently and more power in this system?
```

Comment:
hard to see the changes with this diff
storgage, thinkg, efficiently were changed below

```suggestion
So that's the general idea here. Do we need to reduce hash sizes for these things? No. Is too much security a waste? It can be. It can be very efficient to reduce the size of these things by half. To reduce downloads, and reduce disk storage by half, and speed things up. That's a meaningful gain. Maybe addresses are a little longer than they need to be, but they're not twice as long as they need to be. This is an interesting idea, so it's something I want to put out there. It's something we should think about. Maybe there are other places that people are working on where there's proof of work happening. The Bitcoin blockchain keeps going on, doing enormous amounts of work and maybe we're wasting some of that work. Maybe we can use it to increase efficiency. If you're been looking at the news in the last 6 months since Bitcoin's been going up, a lot of people think Bitcoin is really wasteful and it's pointless. I don't think they're right about it. But are we using this proof of work? It's happening. Are we taking full advantage of this? Can we leverage it for more efficiency and more power in this system?
```